examples/plot_ransac.py

```diff
@@ -0,0 +1,125 @@
+"""
+===============================
+Detect bad sensors using RANSAC
+===============================
+
+This example demonstrates how to use RANSAC [1] from the PREP pipeline to
+detect bad sensors and repair them. Note that this implementation in
+:mod:`autoreject` [2] is an extension of the original implementation and
+works for MEG sensors as well.
+
+[1] Bigdely-Shamlo, Nima, et al.
+    "The PREP pipeline: standardized preprocessing for large-scale EEG
+    analysis." Frontiers in neuroinformatics 9 (2015).
+[2] Mainak Jas, Denis Engemann, Yousra Bekhti, Federico Raimondo, and Alexandre Gramfort,
+    "Autoreject: Automated artifact rejection for MEG and EEG." arXiv preprint
+    arXiv:1612.08194, 2016.
+"""
+
+# Author: Mainak Jas <mainak.jas@telecom-paristech.fr>
+# License: BSD (3-clause)
+
+###############################################################################
+# For the purposes of this example, we shall use the MNE sample dataset.
+# Therefore, let us make some MNE related imports.
+
+import mne
+from mne import io
+from mne import Epochs
+from mne.datasets import sample
+
+###############################################################################
+# Let us now read in the raw `fif` file for MNE sample dataset.
+
+data_path = sample.data_path()
+raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
+raw = io.read_raw_fif(raw_fname, preload=True)
+
+###############################################################################
+# We can then read in the events
+
+event_fname = data_path + ('/MEG/sample/sample_audvis_filt-0-40_raw-'
+                           'eve.fif')
+event_id = {'Auditory/Left': 1}
+tmin, tmax = -0.2, 0.5
+
+events = mne.read_events(event_fname)
+
+###############################################################################
+# And pick MEG channels for repairing. Currently, :mod:`autoreject` can repair
+# only one channel type at a time.
+
+raw.info['bads'] = []
+picks = mne.pick_types(raw.info, meg='grad', eeg=False, stim=False, eog=False,
+                       include=[], exclude=[])
+
+###############################################################################
+# Now, we can create epochs. The ``reject`` params will be set to ``None``
+# because we do not want epochs to be dropped when instantiating
+# :class:`mne.Epochs`.
+
+raw.info['projs'] = list()  # remove proj, don't proj while interpolating
+epochs = Epochs(raw, events, event_id, tmin, tmax,
+                picks=picks, baseline=(None, 0), reject=None,
+                verbose=False, detrend=0, preload=True)
+
+###############################################################################
+# First, we set up the function to compute the sensor-level thresholds.
+from autoreject import Ransac
+from autoreject.utils import interpolate_bads
+
+ransac = Ransac(ch_type='meg', verbose='tqdm')
+ransac.fit_predict(epochs)
+epochs_clean = epochs.copy()
```

Comment:
API is weird. The output of a predict is usually used. It should be a fit or just not operate in place.